libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts

```diff
@@ -199,6 +215,8 @@ export class OAuthPrompt extends Dialog {
             // Return recognized value or re-prompt
             if (isValid) {
                 return await dc.endDialog(recognized.value);
+            } else if (isMessage && this.settings.endOnInvalidMessage) {
```

Comment:
I always prefer to return early and use guard clauses in the code to keep it clean and easier to read, but some may argue having one entry/exit is a better practice.

Reply:
changed, thanks @munozemilio